File: tuethu/hello/hello.cpp
#include <iostream>

int x;  // Global Variables are BAD!!!!!

void printArray(int array[], int size)
{
    int i = 0;
    x = 42;
    while (i < size) {
        std::cout << "array[" << i << "] = " << array[i] << "\n";
        i++;
    }
    //array[3] = 1234;
}

int main(int argc, char *argv[])
{
    std::cout << "Hello World!\n";
    for (int i=0; i<argc; i++) {
        std::cout << "arg: ("+std::string(argv[i])+")\n";
    }
    int vals[10];
    for (int i=0; i < 10; i++) {
        vals[i] = i+20;
    }
    //std::cout << "array size is " << vals.length << "\n";
    printArray(vals,10);
    //std::cout << "vals[3]=" << vals[3] << "\n";
    std::cout << "x is " << x << "\n";
    return 0;
}

File: tuethu/java/Hello.java

public class Hello
{

static void printArray(int array[])
{
    System.out.println("Array size is " + array.length);
    int i = 0;
    while (i < 10) {
        System.out.println("array["+i+"] = "+array[i]);
        i++;
    }
    array[3] = 1234;
}

public static void main(String args[])
{
    System.out.println("Hello World!");
    for (String arg: args) {
        System.out.println("arg: ("+arg+")");
    }
    int[] vals;
    vals = new int[10];
    for (int i=0; i < 10; i++) {
        vals[i] = i+20;
    }
    printArray(vals);
    System.out.println("vals[3] = "+vals[3]);
}

} // end class

